server/app/services/program/ActiveAndDraftPrograms.java

```diff
@@ -45,31 +46,36 @@ public static ActiveAndDraftPrograms buildFromCurrentVersionsUnsynced(
     return new ActiveAndDraftPrograms(repository, Optional.empty());
   }
 
+  private ImmutableMap<String, ProgramDefinition> mapNameToProgramWithFilter(
+      VersionRepository repository,
+      Optional<ProgramService> service,
+      VersionModel versionModel,
+      Optional<DisplayMode> excludeDisplayMode) {
+    return repository.getProgramsForVersion(checkNotNull(versionModel)).stream()
+        .map(
+            program ->
+                service.isPresent()
```

Comment:
**suggestion:** Could we use ifPresentOrElse here and below? https://docs.oracle.com/javase%2F9%2Fdocs%2Fapi%2F%2F/java/util/Optional.html#ifPresentOrElse-java.util.function.Consumer-java.lang.Runnable-

Reply:
I attempted to use `ifPresentOrElse()` in the filter one for about about an hour and couldn't really do so. Maybe I'm missing something. If you think it's doable I can give it more tries in a future PR.

Reply:
I'll take a look but sgtm on moving forward for now!

Reply:
I actually think `ifPresent` wouldn't work in this case because it returns a void.  We could use map to do something like:

`.map(
            program ->
                service.map(svc -> getFullProgramDefinition(svc, program.id)).orElse(program.getProgramDefinition())`

But I think this is fine as is. Sorry for the confusion!